src/runtime_src/core/tools/xbmgmt2/xbmgmt

```diff
@@ -60,89 +62,92 @@ declare -A COMMANDS=( \
+	    XRT_PROG=xbmgmt2
+	    shift
+	    ;;
+	# Indicates that the classic version of xbutil should be used
+	-old|--old)
+	    XRT_PROG=xbmgmt
+	    shift
+	    ;;
+	# Copy the options the remaining options
+	*)
+	    XRTWRAP_PROG_ARGS[$XRTWARP_PROG_ARGS_size]="$1"
+	    XRTWARP_PROG_ARGS_size=$(($XRTWARP_PROG_ARGS_size + 1))
+	    shift
+	    ;;
+    esac
 done
 
-# If user did not specify -new/--new then attempt to determine if user
+# Determine which version of xbmgmt should be used
+# Check to see environment variable is NOT set
+if [ "${XRT_TOOLS_NEXTGEN,,}" = "true" ]; then
+    XRT_PROG=xbmgmt2
+fi
+
+# If user did not specify -old/--old then attempt to determine if user
 # wants to use classic or new by peeking inside the command line
 # Ideally with bash 'nameref' the case/esac would not be necessary and
 # we should be able to deref in one step using helper declaration like
 # declare -n PROG=${COMMANDS[${XRTWRAP_PROG_ARGS[0]}]}
 # However RHEL 7.X still uses bash 4.2 which does not support nameref :-(
 
-if [ ${XRT_PROG} == "xbmgmt" ]; then
-  PROG=${COMMANDS[${XRTWRAP_PROG_ARGS[0]}]}
-  case ${PROG} in
-    XBMGMT | XBMGMT2)
-      # Get named reference to the value pointed by the string returned
-      XRT_PROG=${!PROG}
-      ;;
-    RSTOPT)
-      # We have an ambiguous sub-command, use the next token/switch to
-      # disambiguate by a second look up
-      if [ ${#XRTWRAP_PROG_ARGS[@]} -gt 1 ]; then
-        PROG=${RSTOPT[${XRTWRAP_PROG_ARGS[1]}]:-${DEFAULT}}
-        XRT_PROG=${!PROG}
-      else
-        # User did not provide switch to the sub-command so we cannot
-        # disabmiguate, default to DEFAULT
-        XRT_PROG=${DEFAULT}
-      fi
-      ;;
-    *)
-      # Bogus sub-command which failed lookup
-      XRT_PROG=${DEFAULT}
-      ;;
-  esac
+if [[ -z ${XRT_PROG} ]] && [[ ${#XRTWRAP_PROG_ARGS[@]} -gt 0 ]]; then
+    PROG=${COMMANDS[${XRTWRAP_PROG_ARGS[0]}]}
+    case ${PROG} in
+	XBMGMT | XBMGMT2)
+	# Non ambiguous command line argument, utility already resolved
+	;;
+	RSTOPT)
+	    # We have an ambiguous sub-command, use the next token/switch to
+	    # disambiguate by a second look up
+	    PROG="DEFAULT"
+	    if [ ${#XRTWRAP_PROG_ARGS[@]} -gt 1 ]; then
+		PROG=${RSTOPT[${XRTWRAP_PROG_ARGS[1]}]:-${DEFAULT}}
+	    fi
+	    ;;
+	*)
+	    # Bogus sub-command which failed lookup
+	    PROG="DEFAULT"
+	    ;;
+    esac
+    # Get named reference to the value pointed by the string returned
+    XRT_PROG=${!PROG}
+fi
+
+# Default to the new gen tool if we still could not disambuguate the choice
+if [[ -z ${XRT_PROG} ]]; then
+    XRT_PROG=${DEFAULT}
 fi
 
 # Friendly message
-if [ ${XRT_PROG} == "xbmgmt2" ]; then
-  echo "-----------------------------------------------------"
-  echo " Invoking next generation of the xbmgmt application "
-  echo "-----------------------------------------------------"
+if [[ ${XRT_PROG} == "xbmgmt2" ]]; then
+    echo "-----------------------------------------------------"
```

Comment:
@rozumx, should we remove this as this causes issues when the user's script tries to parse a json file? I believe you have a CR for that

Reply:
Sonal already made the changes.